core/commands/get.go

```diff
@@ -166,12 +166,20 @@ func getCompressOptions(req cmds.Request) (int, error) {
 	return gzip.NoCompression, nil
 }
 
-func get(ctx context.Context, node *core.IpfsNode, p string, compression int) (io.Reader, error) {
+func get(ctx context.Context, node *core.IpfsNode, p string, compression int) (io.Reader, uint64, error) {
 	pathToResolve := path.Path(p)
 	dagnode, err := core.Resolve(ctx, node, pathToResolve)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
-	return utar.NewReader(pathToResolve, node.DAG, dagnode, compression)
+	reader, err := utar.NewReader(pathToResolve, node.DAG, dagnode, compression)
+	ns, err := dagnode.Stat()
+	if err != nil {
+		return nil, 0, err
+	}
+
+	length := uint64(ns.CumulativeSize)
```

Comment:
yeah, the ipfs size is the real amount of data moved. the unixfs size may be smaller, but could even be larger (say if i have 1000 links to the same, huge object). whatever we pick we should use it correctly (i.e. the >100% i think is because they're perhaps mixed)